applications/plugins/CGALPlugin/FindCGAL.cmake

```diff
@@ -1,103 +1,129 @@
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+# FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+# COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+#=============================================================================
+
+message(STATUS "Checking for package 'CGAL'")
+
+# Blank out CGAL_FIND_VERSION temporarily or else find_package(CGAL ...)
+# (below) will fail.
+set(CGAL_FIND_VERSION_TMP ${CGAL_FIND_VERSION})
+set(CGAL_FIND_VERSION "")
+
+# Call CGAL supplied CMake script
+find_package(CGAL
+  HINTS
+  ${CGAL_DIR}
+  $ENV{CGAL_DIR}
+   /usr/lib/${CMAKE_LIBRARY_ARCHITECTURE}/cmake
+  PATH_SUFFIXES lib cmake/modules lib/cmake lib/CGAL)
+
+# Restore CGAL_FIND_VERSION
+set(CGAL_FIND_VERSION ${CGAL_FIND_VERSION_TMP})
+
+if (CGAL_FIND_VERSION)
+  # Check if version found is >= required version
+  if (NOT "${CGAL_VERSION}" VERSION_LESS "${CGAL_FIND_VERSION}")
+    set(CGAL_VERSION_OK TRUE)
+  endif()
+else()
+  # No specific version of CGAL is requested
+  set(CGAL_VERSION_OK TRUE)
 endif()
 
-if ( CGAL_DIR )
+# Add flag to fix bug in CGAL 4.1 for Intel compilers. See
+# https://sympa.inria.fr/sympa/arc/cgal-discuss/2013-01/msg00011.html
+if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Intel")
+  if ("${CGAL_VERSION}" VERSION_GREATER "4.0.2")
+    set(CGAL_DEFINITIONS "-DCGAL_CFG_NO_STATEMENT_EXPRESSIONS")
+  endif()
+endif()
 
-    if ( EXISTS "${CGAL_DIR}/CGALConfig.cmake" )
-        include( "${CGAL_DIR}/CGALConfig.cmake" )
-        set( CGAL_FOUND TRUE )
-    endif()
+# Set variables
+set(CGAL_INCLUDE_DIRS ${CGAL_INCLUDE_DIRS} ${CGAL_3RD_PARTY_INCLUDE_DIRS})
+set(CGAL_LIBRARIES ${CGAL_LIBRARY} ${CGAL_3RD_PARTY_LIBRARIES})
 
+# Add GMP and MPFR libraries if defined by CGAL
+if (GMP_LIBRARIES)
+  set(CGAL_LIBRARIES ${CGAL_LIBRARIES} ${GMP_LIBRARIES})
+endif()
+if (MPFR_LIBRARIES)
+  set(CGAL_LIBRARIES ${CGAL_LIBRARIES} ${MPFR_LIBRARIES})
 endif()
 
-if( NOT CGAL_FOUND)
-    # try to find the paths manually...
-    find_path(CGAL_DIR NAMES include/CGAL/Polyhedron_3.h)
-    find_library(CGAL_LIBRARY NAMES CGAL)
-    find_library(CGAL_ImageIO_LIBRARY NAMES CGAL_ImageIO)
-    if(EXISTS "${CGAL_DIR}/include/CGAL/Polyhedron_3.h" AND EXISTS ${CGAL_LIBRARY})
-        set(CGAL_FOUND TRUE)
-        message(STATUS "CGAL found: ${CGAL_LIBRARY} ${CGAL_DIR}")
-        mark_as_advanced(CGAL_DIR CGAL_LIBRARY)
+# Try compiling and running test program
+if (DOLFIN_SKIP_BUILD_TESTS)
```

Comment:
Oups, my bad, this file was a 100% copy from the Dolfin (fenics) project.

Reply:
I went through quickly, do you confirm I can remove the whole block, the if but also the elseif section. They are not relevant for us?

Reply:
I think we could keep the test, ie remove the current `if` and change the `elseif` to a `if`